lib/Reference/ProfilePickerReferenceProvider.php

```diff
@@ -0,0 +1,188 @@
+	private ?string $userId;
+	private IL10N $l10n;
+	private IURLGenerator $urlGenerator;
+	private IUserManager $userManager;
+	private IAccountManager $accountManager;
+
+	public function __construct(
+		IL10N $l10n,
+		IURLGenerator $urlGenerator,
+		IUserManager $userManager,
+		IAccountManager $accountManager,
+		?string $userId
+	) {
+		$this->userId = $userId;
+		$this->l10n = $l10n;
+		$this->urlGenerator = $urlGenerator;
+		$this->userManager = $userManager;
+		$this->accountManager = $accountManager;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getId(): string	{
+		return 'profile_picker';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getTitle(): string {
+		return $this->l10n->t('Profile picker');
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getOrder(): int	{
+		return 10;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getIconUrl(): string {
+		return $this->urlGenerator->imagePath(Application::APP_ID, 'profile-dark.svg');
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function matchReference(string $referenceText): bool {
+		return $this->getObjectId($referenceText) !== null;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function resolveReference(string $referenceText): ?IReference {
+		if (!$this->matchReference($referenceText)) {
+			return null;
+		}
+
+		$userId = $this->getObjectId($referenceText);
+		$user = $this->userManager->get($userId);
+		if ($user !== null) {
+			$reference = new Reference($referenceText);
+
+			$userDisplayName = $user->getDisplayName();
+			$userEmail = $user->getEMailAddress();
+			$userAvatarUrl = $this->urlGenerator->linkToRouteAbsolute('core.avatar.getAvatar', ['userId' => $userId, 'size' => '64']);
+
+			$bio = $this->accountManager->getAccount($user)->getProperty(IAccountManager::PROPERTY_BIOGRAPHY);
+			$bio = $bio->getScope() !== IAccountManager::SCOPE_PRIVATE ? $bio->getValue() : null;
+			$headline = $this->accountManager->getAccount($user)->getProperty(IAccountManager::PROPERTY_HEADLINE);
+			$location = $this->accountManager->getAccount($user)->getProperty(IAccountManager::PROPERTY_ADDRESS);
+			$website = $this->accountManager->getAccount($user)->getProperty(IAccountManager::PROPERTY_WEBSITE);
+			$organisation = $this->accountManager->getAccount($user)->getProperty(IAccountManager::PROPERTY_ORGANISATION);
+			$role = $this->accountManager->getAccount($user)->getProperty(IAccountManager::PROPERTY_ROLE);
+
+			// for clients who can't render the reference widgets
+			$reference->setTitle($userDisplayName);
+			$reference->setDescription($userEmail ?? $userDisplayName);
+			$reference->setImageUrl($userAvatarUrl);
+
+			// for the Vue reference widget
+			$reference->setRichObject(
+				self::RICH_OBJECT_TYPE,
+				[
+					'user_id' => $userId,
+					'title' => $userDisplayName,
+					'subline' => $userEmail ?? $userDisplayName,
+					'email' => $userEmail,
+					'bio' => isset($bio) && $bio !== '' ? substr_replace($bio, '...', 80, strlen($bio)) : null,
```

Comment:
I think we should use a multibyte-safe string function, e.g. https://www.php.net/manual/en/function.mb-substr.php and then append `…`. `substr_replace` and `strlen` do not work well with multibytes/emojis.